### ISSUE TITLE
Option to use rustls and/or disable tls support entirely

### ISSUE THREAD

**Issue by jamesbirtles** [KallDrexx/mmids#32]
Building mmids requires openssl to build making it a bit more annoying to build when you don't want/need tls support. It would be nice if tls was an optional feature, or if the native tls could be swapped out for rustls for a fully rust based build.

**Comment by KallDrexx** [comment]
I don't remember why I chose native-tls instead of rusttls when originally.  It might have been docs or ease of implementation related but I don't remember at this point.

I can experiment with swapping to rusttls and see what the breadth of changes look to be.

I'm hesitant to make tls optional as that makes things a bit more complicated, including ensuring that the rtmp_receive workflow step knows when to allow the rtmps flag and when not to.  Currently the actual networking is pretty extracted from that level.

**Comment by KallDrexx** [comment]
FYI, I almost have rustls swapped in.  Just need to do some final checking before opening up a PR.

**Comment by KallDrexx** [comment]
Actually ignore the last message,  `rustls` seems less capable, as it can't handle encrypted pem without additional hoops (and other crypto crates).  It's not clear to me why openssl is an issue enough to really figure out how to figure out how to make rusttls work adequately.

I'll have to think about it a bit more.  Making TLS optional isn't going to be easy as the application scales out, because it's going to be required for things like `rtmp_push`, which will need TLS support in order to support pushes to RTMPS destinations, such as Facebook.

**Comment by jamesbirtles** [comment]
Fair enough, I’ll just suck it up and install openssl-dev! Thanks for looking into it so quickly though 🙂
